R/AbstractGraphReporter.R

```diff
@@ -94,11 +94,21 @@ AbstractGraphReporter <- R6::R6Class(
             )
 
             # Round the double columns to three digits for formatting reasons
-            numCols <- names(which(unlist(lapply(tableObj$x$data, is.double))))
+
+            # what are the double columns (but still may contain values with all integers)?
+
+            numDCols <- names(which(unlist(lapply(tableObj$x$data, is.double))))
+
+            # Check if each individual value in a double column is an integer,
+            # If at least one of the values is not an integers, then we want to round only those columns
```

Comment:
Did you observe this condition while testing? I believe that using a `data.frame` (which we are by using `data.table`) guarantees that all values in a column are the same type. Why was this check necessary?

Are you saying that you are checking for columns which have all values like `c(1.0, 2.0, 3.0)` that _could be coerced to integer_? If so I don't think that is necessary and I think it could lead to inconsistent behavior.

I recommend removing this line unless you tell me I've misunderstood something.

Reply:
Agreed, all columns are indeed the same type. The challenge is not all double/numeric columns have values with decimals such as 1.1, 1.3, etc and are actually integers so I'm checking as you said if they could be coerced to integers (we don't want to round these columns because it just adds three 0s). The previous behavior was simply rounding all double columns.